src/main/java/com/microsoft/aad/msal4j/AadInstanceDiscoveryProvider.java

```diff
@@ -129,34 +129,59 @@ private static AadInstanceDiscoveryResponse sendInstanceDiscoveryRequest(URL aut
                                                                              MsalRequest msalRequest,
                                                                              ServiceBundle serviceBundle) {
 
-        String region = StringHelper.EMPTY_STRING;
         IHttpResponse httpResponse = null;
-
-        //If the autoDetectRegion parameter in the request is set, attempt to discover the region
-        if (msalRequest.application().autoDetectRegion()) {
-            region = discoverRegion(msalRequest, serviceBundle);
+        String providedRegion = msalRequest.application().azureRegion();
+        String detectedRegion = null;
+        int regionOutcome = 0;
+        String regionToUse = null;
+
+        //If a region was provided by a developer or they set the autoDetectRegion parameter,
+        // attempt to discover the region and set telemetry info based on the outcome
+        if (providedRegion != null) {
+            detectedRegion = discoverRegion(msalRequest, serviceBundle);
```

Comment:
why do we discover region if it was provided and auto detect might not be enabled?

Reply:
The region design doc specifies that if the region API is used at all, then the library should attempt to discover the region. From the spec: "Region feature is opt-in ... if developer opts in, region auto-detection should always happen and telemetry MUST be used to send back region data,"

This is entirely for telemetry: half of the 'region outcome' telemetry values deal with the dev setting the region, and autodetect succeeding/failing/having a mismatch with the provided region. So, even if it's not needed when the region is provided, I guess it's considered an important piece of telemetry info.